kernel/ramfs.c

```diff
@@ -534,8 +534,10 @@ static int _path_to_inode_recursive(
     {
         for (size_t i = 0; i < ntoks; i++)
         {
-            inode_t* p;
+            if (!S_ISDIR(parent->mode))
```

Comment:
When would this happen?

Reply:
If any intermediate path component is a file. For eg: /reg-file/file2. On linux this returns on ENOTDIR, mystikos was returning an ENOENT.